src/BLI.jl

```diff
@@ -9,3 +13,53 @@ end
 function R2Calc(R::Real, Kdis::Real, tps)
     return R * exp.(-Kdis * tps)
 end
+
+
+function importData(cond)
+    df = CSV.read(cond)
+    conc = df[1,2:end]
+    tps = df[5:end,1]
+    measVal = df[5:end,2:end]
+    return conc, parse.(Float64, tps), measVal
+end
+
+function plotBLI(cond)
+    conc, tps, bindData = TAMode.importData(cond)
+    bindData = Matrix(bindData)
+    plot(parse.(tps), parse.(bindData), title=cond, 
+         label=[conc[1] conc[2] conc[3] conc[4] conc[5] conc[6] conc[7] conc[8]], lw=3)
+end
+
+@model BLI(cond) = begin
+    Tshift = 822.2
+    Kon ~ Normal(6., 0.5)
+    Kdis ~ Normal(2., 1.)
+    
+    conc, tps, bindData = importData(cond)
+    Tshift = tps[1] + 600.01
+    theor_save = 0
+    exp_save = 0
+    for i in 1:length(conc)
+        tBind = tps[tps.<Tshift].-tps[1]
+        L0 = parse(Float64, conc[i])
```

Comment:
Also don't prepare or parse the data here. Get it all tidied up outside of the model, then give the model a prepared array in exactly the structure it needs.

Reply:
fixed it in the recent commit